packages/std/src/ado_contract/permissioning.rs

```diff
@@ -820,6 +862,81 @@ mod tests {
         assert!(res.is_ok());
     }
 
+    #[test]
+    fn test_permission_start_time() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        env.block.height = 0;
+        let action = "action";
+        let actor = "actor";
+        let contract = ADOContract::default();
+        let start_time = 2;
+        let start = Some(Expiry::AtTime(MillisecondsExpiration::from_seconds(
+            start_time,
+        )));
+
+        env.block.time = MillisecondsExpiration::from_seconds(0).into();
+        contract
+            .owner
+            .save(deps.as_mut().storage, &Addr::unchecked("owner"))
+            .unwrap();
+
+        ADOContract::default()
+            .permission_action(action, deps.as_mut().storage)
+            .unwrap();
+
+        // Test Whitelist with start time
+        let permission = Permission::Local(LocalPermission::Whitelisted {
+            start: start.clone(),
+            expiration: None,
+        });
+        ADOContract::set_permission(deps.as_mut().storage, action, actor, permission).unwrap();
+
+        // Before start time - should error
+        let res = contract.is_permissioned(deps.as_mut(), env.clone(), action, actor);
+        assert!(res.is_err());
+
+        // At start time - should succeed
```

Comment:
Let's rewrite this in a case based structure using `rstest` so it's easier to read/maintain.